skills/csharp/experimental/itsmskill/Dialogs/Teams/TicketTaskModule/CreateTicketTeamsImplementation.cs

```diff
@@ -0,0 +1,138 @@
+
+    /// <summary>
+    /// CreateTicketTeamsImplementation Handles OnFetch and OnSumbit Activity for TaskModules
+    /// </summary>
+    [TeamsInvoke(FlowType = nameof(TeamsFlowType.CreateTicket_Form))]
+    public class CreateTicketTeamsImplementation : ITeamsTaskModuleHandler<TaskModuleResponse>
+    {
+        private readonly IStatePropertyAccessor<SkillState> _stateAccessor;
+        private readonly ConversationState _conversationState;
+        private readonly BotSettings _settings;
+        private readonly BotServices _services;
+        private readonly IServiceManager _serviceManager;
+
+        public CreateTicketTeamsImplementation(
+             IServiceProvider serviceProvider)
+        {
+            _conversationState = serviceProvider.GetService<ConversationState>();
+            _settings = serviceProvider.GetService<BotSettings>();
+            _services = serviceProvider.GetService<BotServices>();
+            _stateAccessor = _conversationState.CreateProperty<SkillState>(nameof(SkillState));
+            _serviceManager = serviceProvider.GetService<IServiceManager>();
+        }
+
+        // Handle Fetch
+        public async Task<TaskModuleResponse> OnTeamsTaskModuleFetchAsync(ITurnContext context, CancellationToken cancellationToken)
+        {
+            return new TaskModuleResponse()
+            {
+                Task = new TaskModuleContinueResponse()
+                {
+                    Value = new TaskModuleTaskInfo()
+                    {
+                        Title = "ImpactTracker",
+                        Height = "medium",
+                        Width = 500,
+                        Card = new Attachment
+                        {
+                            ContentType = AdaptiveCard.ContentType,
+                            Content = TicketDialogHelper.CreateIncidentAdaptiveCard()
+                        }
+                    }
+                }
+            };
+        }
+
+        // Handle Submit True
+        public async Task<TaskModuleResponse> OnTeamsTaskModuleSubmitAsync(ITurnContext context, CancellationToken cancellationToken)
+        {
+            var state = await _stateAccessor.GetAsync(context, () => new SkillState());
+            var accessToken = state.AccessTokenResponse.Token;
+            var activityValueObject = JObject.FromObject(context.Activity.Value);
+
+            var isDataObject = activityValueObject.TryGetValue("data", StringComparison.InvariantCultureIgnoreCase, out JToken dataValue);
+            JObject dataObject = null;
+            if (isDataObject)
+            {
+                dataObject = dataValue as JObject;
+
+                // Get Title
+                var title = dataObject.GetValue("IncidentTitle");
+
+                // Get Description
+                var description = dataObject.GetValue("IncidentDescription");
+
+                // Get Urgency
+                var urgency = dataObject.GetValue("IncidentUrgency");
```

Comment:
is it possible title, description and urgency are empty? should we check before using them?